src/Network/HTTPClient/Capability/ICanSendHttpRequests.php

```diff
@@ -50,72 +51,75 @@ public function fetch(string $url, int $timeout = 0, string $accept_content = ''
+	 * @param string $url            URL to get
+	 * @param string $accept_content supply Accept: header with 'accept_content' as the value
+	 * @param array  $opts           (optional parameters) associative array with:
+	 *                                'accept_content' => (string array) supply Accept: header with 'accept_content' as the value (overrides default parameter)
+	 *                                'timeout' => int Timeout in seconds, default system config value or 60 seconds
+	 *                                'cookiejar' => path to cookie jar file
+	 *                                'header' => header array
 	 *
 	 * @return ICanHandleHttpResponses
 	 */
-	public function get(string $url, array $opts = []): ICanHandleHttpResponses;
+	public function get(string $url, string $accept_content = HttpClientAccept::DEFAULT, array $opts = []): ICanHandleHttpResponses;
 
 	/**
 	 * Sends a HTTP request to a given url
 	 *
-	 * @param string $method A HTTP request
-	 * @param string $url    Url to send to
-	 * @param array  $opts   (optional parameters) associative array with:
-	 *                       	 'body' => (mixed) setting the body for sending data
-	 *                           'accept_content' => (string array) supply Accept: header with 'accept_content' as the value
-	 *                           'timeout' => int Timeout in seconds, default system config value or 60 seconds
-	 *                           'cookiejar' => path to cookie jar file
-	 *                           'header' => header array
-	 *                           'content_length' => int maximum File content length
-	 *                           'auth' => array authentication settings
+	 * @param string $method         A HTTP request
+	 * @param string $url            Url to send to
+	 * @param string $accept_content supply Accept: header with 'accept_content' as the value
+	 * @param array  $opts           (optional parameters) associative array with:
+	 *                       	      'body' => (mixed) setting the body for sending data
+	 *                                'accept_content' => (string array) supply Accept: header with 'accept_content' as the value (overrides default parameter)
+	 *                                'timeout' => int Timeout in seconds, default system config value or 60 seconds
+	 *                                'cookiejar' => path to cookie jar file
+	 *                                'header' => header array
+	 *                                'content_length' => int maximum File content length
+	 *                                'auth' => array authentication settings
 	 *
 	 * @return ICanHandleHttpResponses
 	 */
-	public function request(string $method, string $url, array $opts = []): ICanHandleHttpResponses;
+	public function request(string $method, string $url, string $accept_content = HttpClientAccept::DEFAULT, array $opts = []): ICanHandleHttpResponses;
```

Comment:
I suggest to only use `accept` with `get`.

Reply:
Even not for `head`?

At least I found it explicit set for the redir with Magic

Reply:
OK, based on https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD , we should even delete the customized `Accept` header for `head` calls as well ..

Reply:
I don't think so. See this quote:
> The HTTP HEAD method requests the [headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) that would be returned if the HEAD request's URL was instead requested with the HTTP [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) method.

So I think the `head` is a `get` without a body. Which means that it should have the same parameters. And since there are pages that return different content when using different `accept` values, this is really needed.

For example: We are using `head` to determine the `content-type` of a picture that we want to download. Now imagine that the requested resource returns content depending on the requested data. So it possibly will return some webp (that we can't handle at the moment) when requested with the default `accept` value. Then when would possibly treat that resource differently.

Reply:
I'm with @annando, HEAAD requests should have the explicit Accept parameter.